### PR TITLE
Add CCI config for releasing.

### PR DESCRIPTION
**Pull request by RobbieMcKinstry**
This PR gets us most of the way to an automated release + GitHub hosted chart repository. I'd appreciate quick review because I need this to land to make progress on the frontend and documentation, since this PR sets up chart repository hosting.

### CODE REVIEW

**Review by nappelson — COMMENTED** [review]
@RobbieMcKinstry just one quick comment

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      CR_PAGES_BRANCH: "main"
+      CR_PAGES_INDEX_PATH: "docs/index.yaml"
+      CR_INDEX_PATH: "docs/index.yaml"
+
+    steps:
+      - checkout
+      - github/setup
+      - run:
+          name: Configure git identitiy.
+          command: git config --global user.name "CircleCI" && git config --global user.email "eng@ottertune.com"
```
[the remark this answers is not in the document]
Comment (nappelson) [reply .circleci/config.yml:37]: ah okay cool. No that's fine!

**Review by nappelson — APPROVED** [review]
@RobbieMcKinstry LGTM! You may have already been planning on doing this but a README about what is going on in this repo would be helpful, specifically how to release new versions, how users consume the helm chart, etc. Could potentially even include some of what you wrote in slack.

**Comment by RobbieMcKinstry** [comment]
@nappelson Yep working on it currently.

**Pull request merged**